map/autoware_pointcloud_divider/src/pcd_divider.cpp

```diff
@@ -0,0 +1,482 @@
+
+template <class PointT>
+void PCDDivider<PointT>::mergeAndDownsample(
+  const std::string & dir_path, std::list<std::string> & pcd_list, size_t total_point_num)
+{
+  PclCloudPtr new_cloud(new PclCloudType);
+
+  new_cloud->reserve(total_point_num);
+
+  // Merge all PCDs that belong to the specified segment to a single segment point cloud
+  for (auto & fname : pcd_list) {
+    PclCloudType seg_cloud;
+
+    if (pcl::io::loadPCDFile(fname, seg_cloud)) {
+      RCLCPP_ERROR(logger_, "Error: Failed to open a PCD file at %s", fname.c_str());
+      rclcpp::shutdown();
+      exit(EXIT_FAILURE);
+    }
+
+    for (auto & p : seg_cloud) {
+      new_cloud->push_back(p);
+    }
+  }
+
+  // Downsample if needed
+  if (leaf_size_ > 0) {
+    VoxelGridFilter<PointT> vgf;
+    PclCloudPtr filtered_cloud(new PclCloudType);
+
+    vgf.setResolution(leaf_size_);
+    vgf.filter(*new_cloud, *filtered_cloud);
+
+    new_cloud = filtered_cloud;
+  }
+
+  // Save the new_cloud
+  // Extract segment name only (format gx_gy)
+  int end_name, start_name;
+
+  for (end_name = dir_path.size() - 1; end_name >= 0 && dir_path[end_name] == '/'; --end_name) {
+  }
+  for (start_name = end_name; start_name >= 0 && dir_path[start_name] != '/'; --start_name) {
+  }
+
+  std::string seg_name_only = dir_path.substr(start_name + 1, end_name - start_name);
+
+  // Parse the @seg_path to get the indices of the segment
+  auto underbar_pos = seg_name_only.rfind("_");
+  int gx = std::stoi(seg_name_only.substr(0, underbar_pos));
+  int gy = std::stoi(seg_name_only.substr(underbar_pos + 1));
+
+  grid_set_.insert(GridInfo<2>(gx, gy));
+
+  // Construct the path to save the new_cloud
+  std::string save_path;
+
+  // If save large pcd was turned on, create a folder to contain segment pcds
+  if (use_large_grid_) {
+    int large_gx = static_cast<int>(std::floor(static_cast<float>(gx) / g_grid_size_x_));
+    int large_gy = static_cast<int>(std::floor(static_cast<float>(gy) / g_grid_size_y_));
+    std::string large_folder =
+      output_dir_ + "/" + std::to_string(large_gx) + "_" + std::to_string(large_gy) + "/";
+
+    // Create a new folder for the large grid
+    util::make_dir(large_folder);
+
+    save_path = large_folder + file_prefix_ + "_" + seg_name_only + ".pcd";
+  } else {
+    save_path = output_dir_ + "/" + file_prefix_ + "_" + seg_name_only + ".pcd";
+  }
+
+  // Save the merged (filtered) cloud
+  if (pcl::io::savePCDFileBinary(save_path, *new_cloud)) {
+    RCLCPP_ERROR(logger_, "Error: Failed to save a point cloud at %s", save_path.c_str());
+    rclcpp::shutdown();
+    exit(EXIT_FAILURE);
+  }
+
+  // Delete the folder containing the segments
+  util::remove(dir_path);
+}
+
+template <class PointT>
+std::string PCDDivider<PointT>::makeFileName(const GridInfo<2> & grid) const
+{
+  std::string file_name = output_dir_;
+
+  file_name = file_name + file_prefix_ + "_";
+  file_name = file_name + std::to_string(grid.ix) + "_" + std::to_string(grid.iy) + ".pcd";
```

Comment:
```suggestion
  std::string file_name = file_prefix_ + "_" + std::to_string(grid.ix) + "_" + std::to_string(grid.iy) + ".pcd";
```
not need output_dir name.
https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_loader#metadata-structure

Reply:
@YamatoAndo I removed that output_dir_.